### PR TITLE
Create searchable select components

### PR DESCRIPTION
**Pull request by ragnarob**
Much copied from Select. Keyboard navigation not implemented. Outside clicks to close also not implemented yet. Later.

### CODE REVIEW

File: app/components/SearchableSelect/SearchableSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+          <div
+            className="z-40 px-3 whitespace-nowrap text-gray-700 hover:cursor-default"
+            onClick={() => onSelected(null)}
+          >
+            No results
+          </div>
+        )}
+      </div>
+
+      <input type="text" name={name} value={value} hidden />
```
Comment (ragnarob) [inline app/components/SearchableSelect/SearchableSelect.tsx:222]: To make it work natively in forms

File: app/components/Select/Select.tsx [hunk 2]
```diff
@@ -156,6 +158,8 @@ export default function Select({
           </div>
         ))}
       </div>
+
+      <input type="text" name={name} value={value} hidden />
```
Comment (ragnarob) [inline app/components/Select/Select.tsx:162]: To make it work natively in forms

File: app/components/SearchableSelect/SearchableSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      if (minWidth > (maxWidth as number)) {
+        setWidth(maxWidth as number);
+      } else {
+        setMinWidth(maxChildWidth);
+      }
+
+      return true;
+    } else {
+      return false;
+    }
```
Comment (ErikMelton) [inline app/components/SearchableSelect/SearchableSelect.tsx:74]: Redundant else clause

File: app/components/SearchableSelect/SearchableSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  }
+
+  function onSelected(clickedValue: any) {
+    setSearchText('');
+    onChange(clickedValue);
+    setIsOpen(false);
+  }
+
+  const convertedOptions = useMemo<keyValOptions[]>(() => {
+    if (!options || !options.length) {
```
Comment (ErikMelton) [inline app/components/SearchableSelect/SearchableSelect.tsx:112]:
```suggestion
    if (!options?.length) {
```

**Review by ErikMelton — CHANGES_REQUESTED** [review]
Small code cleanliness tweaks

**Review by ErikMelton — APPROVED** [review]
LGTM :shipit:

**Pull request merged**